src/main/java/com/blackducksoftware/integration/hub/nexus/util/ParallelEventProcessor.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * hub-nexus
+ *
+ * 	Copyright (C) 2018 Black Duck Software, Inc.
+ * 	http://www.blackducksoftware.com/
+ *
+ * 	Licensed to the Apache Software Foundation (ASF) under one
+ * 	or more contributor license agreements. See the NOTICE file
+ * 	distributed with this work for additional information
+ * 	regarding copyright ownership. The ASF licenses this file
+ * 	to you under the Apache License, Version 2.0 (the
+ * 	"License"); you may not use this file except in compliance
+ * 	with the License. You may obtain a copy of the License at
+ *
+ * 	http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * 	Unless required by applicable law or agreed to in writing,
+ * 	software distributed under the License is distributed on an
+ * 	"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * 	KIND, either express or implied. See the License for the
+ * 	specific language governing permissions and limitations
+ * 	under the License.
+ */
+package com.blackducksoftware.integration.hub.nexus.util;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.TimeUnit;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.slf4j.Logger;
+import org.sonatype.sisu.goodies.common.Loggers;
+
+import com.blackducksoftware.integration.hub.nexus.event.handler.HubEventHandler;
+
+@Named
+@Singleton
+public class ParallelEventProcessor {
+    private final Logger logger = Loggers.getLogger(getClass());
+    private ExecutorService executorService;
+
```

Comment:
I would say add a method called initializeExecutorService() that will encapsulare calling the createExecutorService method and the setExecutorService method inside of the method.

Also at the start of the initializeExecutorService method I would check to see if the executor service is not null.  If it isn't null I would call shutdownNow to release any resources of the previous executor service.  Then continue to create and set the executor service.   The tasks would call the initializeExecutorService() method rather than call the createExecutorService and setExecutorService.  I think the initializeExecutorService would be a cleaner contract for users of the ParallelEventProcessor class.